nym-vpn-x/src-tauri/src/grpc/client.rs

```diff
@@ -75,18 +89,18 @@ impl GrpcClient {
 
     /// Get the Health service client
     #[instrument(skip_all)]
-    pub async fn health(&self) -> Result<HealthClient<Channel>> {
+    pub async fn health(&self) -> Result<HealthClient<Channel>, VpndError> {
         match &self.0 {
             Transport::Http(endpoint) => {
                 HealthClient::connect(endpoint.clone()).await.map_err(|e| {
                     warn!("failed to connect to the daemon: {}", e);
-                    anyhow!("failed to connect to the daemon: {}", e)
+                    VpndError::FailedToConnectHttp(e)
                 })
             }
             Transport::Ipc(socket) => {
                 let channel = get_channel(socket.clone()).await.map_err(|e| {
                     warn!("failed to connect to the daemon: {}", e);
-                    anyhow!("failed to connect to the daemon: {}", e)
+                    VpndError::FailedToConnectIpc(e)
```

Comment:
The name ipc here seems misleading, I would prefer something socket

Reply:
Indeed, `Ipc` is more appropriate as on Unix uds (unix domain sockets) will be used but on Windows it's named pipes

see https://github.com/paritytech/parity-tokio-ipc

Reply:
I think there will likely be a separate enum case to handle windows pipes, not reusing this one which seems to be specifically for local unix sockets

Reply:
So what the point of using `parity-tokio-ipc`?